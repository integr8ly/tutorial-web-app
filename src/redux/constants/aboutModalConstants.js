const ABOUT_MODAL_SHOW = 'ABOUT_MODAL_SHOW';
const ABOUT_MODAL_HIDE = 'ABOUT_MODAL_HIDE';

export { ABOUT_MODAL_SHOW, ABOUT_MODAL_HIDE };
