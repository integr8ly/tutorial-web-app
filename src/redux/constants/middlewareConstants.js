const LIST_WALKTHROUGH = 'LIST_WALKTHROUGH';
const CREATE_WALKTHROUGH = 'CREATE_WALKTHROUGH';
const REMOVE_WALKTHROUGH = 'REMOVE_WALKTHROUGH';
const GET_AMQ_CREDENTIALS = 'GET_AMQ_CREDENTIALS';
const GET_ENMASSE_CREDENTIALS = 'GET_ENMASSE_CREDENTIALS';
const GET_PROVISIONING_USER = 'GET_PROVISIONING_USER';
const GET_CUSTOM_CONFIG = 'GET_CUSTOM_CONFIG';
const PROVISION_SERVICE = 'PROVISION_SERVICE';
const GET_RHMICONFIG_CR = 'GET_RHMICONFIG_CR';

const SERVICE_TYPES = {
  PROVISIONED_SERVICE: 'PROVISIONED_SERVICE'
};

const SERVICE_STATUSES = {
  PROVISIONING: 'PROVISIONING',
  PROVISIONED: 'PROVISIONED',
  DELETING: 'DELETING',
  UNAVAILABLE: 'UNAVAILABLE'
};

export {
  CREATE_WALKTHROUGH,
  REMOVE_WALKTHROUGH,
  LIST_WALKTHROUGH,
  GET_AMQ_CREDENTIALS,
  GET_ENMASSE_CREDENTIALS,
  GET_PROVISIONING_USER,
  GET_CUSTOM_CONFIG,
  PROVISION_SERVICE,
  SERVICE_STATUSES,
  SERVICE_TYPES,
  GET_RHMICONFIG_CR
};
