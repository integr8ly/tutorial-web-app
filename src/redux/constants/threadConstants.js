const GET_THREAD = 'GET_THREAD';
const INIT_THREAD = 'INIT_THREAD';
const UPDATE_THREAD_PROGRESS = 'UPDATE_THREAD_PROGRESS';
const UPDATE_CURRENT_THREAD = 'UPDATE_CURRENT_THREAD';

export { GET_THREAD, INIT_THREAD, UPDATE_THREAD_PROGRESS, UPDATE_CURRENT_THREAD };
