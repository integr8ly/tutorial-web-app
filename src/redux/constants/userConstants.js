const USER_CREATE = 'USER_CREATE';
const USER_DELETE = 'USER_DELETE';
const USER_INFO = 'USER_INFO';
const USER_LOGIN = 'USER_LOGIN';
const USER_LOGOUT = 'USER_LOGOUT';
const USER_STORED_DATA = 'USER_STORED_DATA';
const USER_STORED_DATA_REMOVE = 'USER_STORED_DATA_REMOVE';

export { USER_CREATE, USER_DELETE, USER_INFO, USER_LOGIN, USER_LOGOUT, USER_STORED_DATA, USER_STORED_DATA_REMOVE };
