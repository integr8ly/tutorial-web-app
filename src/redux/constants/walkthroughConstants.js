const LIST_WALKTHROUGH = 'LIST_WALKTHROUGH';
const CREATE_WALKTHROUGH = 'CREATE_WALKTHROUGH';
const REMOVE_WALKTHROUGH = 'REMOVE_WALKTHROUGH';

export { CREATE_WALKTHROUGH, REMOVE_WALKTHROUGH, LIST_WALKTHROUGH };
