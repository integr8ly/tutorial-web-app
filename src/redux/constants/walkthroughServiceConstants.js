const ADD_RESOURCE = 'ADD_RESOURCE';
const REMOVE_RESOURCE = 'REMOVE_RESOURCE';

export { ADD_RESOURCE, REMOVE_RESOURCE };
