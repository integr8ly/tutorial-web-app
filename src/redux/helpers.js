import get from 'lodash.get';

export const getStatusFromResults = results => {
  let status = get(results, 'response.status', results.status);

  if (status === undefined) {
    status = 0;
  }
  return status;
};

export const getMessageFromResults = (results, filterField = null) => {
  const status = get(results, 'response.status', results.status);
  const statusResponse = get(results, 'response.statusText', results.statusText);
  const messageResponse = get(results, 'response.data', results.message);
  const detailResponse = get(results, 'response.data', results.detail);

  if (status < 400 && !messageResponse && !detailResponse) {
    return statusResponse;
  }

  if (status >= 500 || (status === undefined && (messageResponse || detailResponse))) {
    return `${status || ''} Server is currently unable to handle this request. ${messageResponse ||
      detailResponse ||
      ''}`;
  }

  if (typeof messageResponse === 'string') {
    return messageResponse;
  }

  if (typeof detailResponse === 'string') {
    return detailResponse;
  }

  const getMessages = (messageObject, filterKey) => {
    const obj = filterKey ? messageObject[filterKey] : messageObject;
    if (obj) {
      if (typeof obj === 'object') {
        return Object.values(obj);
      }
      if (Array.isArray(obj)) {
        return obj.map(next => {
          if (Array.isArray(next)) {
            return getMessages(next);
          }
          return next;
        });
      }
    }
    return [];
  };
  return getMessages(messageResponse || detailResponse, filterField).join('\n');
};

export const FULFILLED_ACTION = (base = '') => `${base}_FULFILLED`;

export const PENDING_ACTION = (base = '') => `${base}_PENDING`;

export const REJECTED_ACTION = (base = '') => `${base}_REJECTED`;

export const HTTP_STATUS_RANGE = status => `${status}_STATUS_RANGE`;

export const setStateProp = (prop, data, options) => {
  const { state = {}, initialState = {}, reset = true } = options;
  let obj = { ...state };

  if (prop && !state[prop]) {
    console.error(`Error: Property ${prop} does not exist within the passed state.`, state);
  }

  if (reset && prop && !initialState[prop]) {
    console.warn(`Warning: Property ${prop} does not exist within the passed initialState.`, initialState);
  }

  if (reset && prop) {
    obj[prop] = {
      ...state[prop],
      ...initialState[prop],
      ...data
    };
  } else if (reset && !prop) {
    obj = {
      ...state,
      ...initialState,
      ...data
    };
  } else if (prop) {
    obj[prop] = {
      ...state[prop],
      ...data
    };
  } else {
    obj = {
      ...state,
      ...data
    };
  }

  return obj;
};
