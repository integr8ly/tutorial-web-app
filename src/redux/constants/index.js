import * as aboutModalTypes from './aboutModalConstants';
import * as userTypes from './userConstants';
import * as threadTypes from './threadConstants';
import * as walkthroughTypes from './walkthroughConstants';

const reduxTypes = {
  aboutModalTypes,
  userTypes,
  threadTypes
};

export { reduxTypes as default, reduxTypes, aboutModalTypes, userTypes, threadTypes, walkthroughTypes };
