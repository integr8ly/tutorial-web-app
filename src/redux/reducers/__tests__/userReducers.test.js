import { userReducers } from '..';
import { userTypes as types } from '../../constants';
import { REJECTED_ACTION, PENDING_ACTION, FULFILLED_ACTION } from '../../helpers';

describe('UserReducers', () => {
  it('should return the initial state', () => {
    expect(userReducers.initialState).toBeDefined();
  });

  it('should handle all defined error types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = userReducers(undefined, dispatched);

      expect({ type: REJECTED_ACTION(value), result: resultState }).toMatchSnapshot('rejected types');
    });
  });

  it('should handle all defined pending types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: PENDING_ACTION(value)
      };

      const resultState = userReducers(undefined, dispatched);

      expect({ type: PENDING_ACTION(value), result: resultState }).toMatchSnapshot('pending types');
    });
  });

  it('should handle all defined fulfilled types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = userReducers(undefined, dispatched);

      expect({ type: FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot('fulfilled types');
    });
  });
});
