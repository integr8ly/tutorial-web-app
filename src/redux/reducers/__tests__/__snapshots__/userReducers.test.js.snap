// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserReducers should handle all defined error types: rejected types 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_CREATE_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types: rejected types 2`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_DELETE_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types: rejected types 3`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_INFO_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types: rejected types 4`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "loginFailed": true,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGIN_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types: rejected types 5`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGOUT_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types: rejected types 6`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types: rejected types 7`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REMOVE_REJECTED",
}
`;

exports[`UserReducers should handle all defined fulfilled types: fulfilled types 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "userInfo": Object {
        "test": "success",
      },
    },
  },
  "type": "USER_CREATE_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types: fulfilled types 2`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "userInfo": Object {
        "test": "success",
      },
    },
  },
  "type": "USER_DELETE_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types: fulfilled types 3`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_INFO_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types: fulfilled types 4`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": true,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGIN_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types: fulfilled types 5`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGOUT_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types: fulfilled types 6`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": true,
      "storedEmail": undefined,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types: fulfilled types 7`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REMOVE_FULFILLED",
}
`;

exports[`UserReducers should handle all defined pending types: pending types 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "userInfo": Object {},
    },
  },
  "type": "USER_CREATE_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types: pending types 2`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "userInfo": Object {},
    },
  },
  "type": "USER_DELETE_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types: pending types 3`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_INFO_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types: pending types 4`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": true,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGIN_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types: pending types 5`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGOUT_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types: pending types 6`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types: pending types 7`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "email": null,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "username": null,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REMOVE_PENDING",
}
`;
